src/commands/utils/clusters.ts

```diff
@@ -28,6 +29,63 @@ export enum ClusterStartStopState {
     Stopping = 'Stopping'
 }
 
+export interface KubernetesClusterInfo {
+    readonly name: string,
+    readonly kubeconfigYaml: string
+}
+
+export async function getKubernetesClusterInfo(commandTarget: any, cloudExplorer: APIAvailable<CloudExplorerV1>, clusterExplorer: APIAvailable<ClusterExplorerV1>): Promise<Errorable<KubernetesClusterInfo>> {
+
+    // See if this is an AKS cluster, and if so, download the credentials for it.
+    const aksCluster = getAksClusterTreeItem(commandTarget, cloudExplorer);
+    if (succeeded(aksCluster)) {
+        const properties = await longRunning(`Getting properties for cluster ${aksCluster.result.name}.`, () => getClusterProperties(aksCluster.result));
+        if (failed(properties)) {
+            return properties;
+        }
+    
+        const kubeconfigYaml = await getKubeconfigYaml(aksCluster.result, properties.result);
+        if (failed(kubeconfigYaml)) {
+            return kubeconfigYaml;
+        }
+
+        const result = {
+            name: commandTarget.name,
+            kubeconfigYaml: kubeconfigYaml.result
+        };
+
+        return {succeeded: true, result };
+    }
+
+    const configuration = await extension.configuration.v1;
+    if (!configuration.available) {
+        return { succeeded: false, error: 'Unable to retrieve kubeconfig: configuration API unavailable.' };
+    }
+
+    // Not an AKS cluster. This should be a cluster-explorer node. Verify:
+    const explorerCluster = clusterExplorer.api.resolveCommandTarget(commandTarget) as ClusterExplorerV1.ClusterExplorerContextNode;
+    if (explorerCluster === undefined) {
+        return { succeeded: false, error: 'This command should only apply to active cluster nodes.' }
+    }
+
+    const kubeconfigPath = getPath(await configuration.api.getKubeconfigPath());
+    const result = {
+        name: explorerCluster.name,
+        kubeconfigYaml: fs.readFileSync(kubeconfigPath, 'utf8')
+    };
+
+    return { succeeded: true, result };
+}
+
+function getPath(kubeconfigPath: ConfigurationV1.KubeconfigPath): string {
+    switch (kubeconfigPath.pathType) {
+        case 'host':
+            return kubeconfigPath.hostPath;
```

Comment:
💡 Question: This is not changing the existing behaviour? but curious as to how `hostPath` vs `wslPath` was handled before.

Reply:
This is new, because we were never reading the `kubeconfig` file used by cluster explorer before. Until now, we've only been getting the kubeconfig for AKS clusters, but if the user invokes the command from the cluster explorer, we need a different way of getting the kubeconfig (which I grabbed from the [k8s extension docs](https://github.com/vscode-kubernetes-tools/vscode-kubernetes-tools/blob/master/docs/extending/configuration.md#detecting-the-kubernetes-configuration)). I'll put a comment in.